### PR TITLE
Features/#146 regionalization onwind pv

### PR DESCRIPTION
**Pull request by KathiEsterl**
This branch includes the distribution of wind onshore and PV ground mounted power parks. 

Please test the branch and have a look on the code (style, examination code) and also on the results.

Thank you!

### CODE REVIEW

**Comment by ClaraBuettner** [comment]
I tried to run your branch, but the task 'generate_pv_ground_mounted' fails with this error:

```
> 
> [2021-06-24 13:57:25,697] {taskinstance.py:1150} ERROR - 'function' object has no attribute 'cursor'
> Traceback (most recent call last):
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
>     result = task_copy.execute(context=context)
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/airflow/operators/python_operator.py", line 113, in execute
>     return_value = self.execute_callable()
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/airflow/operators/python_operator.py", line 118, in execute_callable
>     return self.python_callable(*self.op_args, **self.op_kwargs)
>   File "/home/clara/test-egon-data/git/eGon-data/src/egon/data/processing/pv_ground_mounted.py", line 849, in regio_of_pv_ground_mounted
>     pv_rora, pv_agri, pv_per_distr, pv_rora_100RE, pv_agri_100RE, pv_per_distr_100RE = run_methodology(con = db.engine, path='', pow_per_area = 0.04, join_buffer = 10, max_dist_hv = 20000, show_map=True)
>   File "/home/clara/test-egon-data/git/eGon-data/src/egon/data/processing/pv_ground_mounted.py", line 521, in run_methodology
>     potentials_rora, potentials_agri = potential_areas(con, join_buffer)
>   File "/home/clara/test-egon-data/git/eGon-data/src/egon/data/processing/pv_ground_mounted.py", line 118, in potential_areas
>     potentials_rora = gpd.GeoDataFrame.from_postgis(sql, con)
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/geopandas/geodataframe.py", line 668, in from_postgis
>     chunksize=chunksize,
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/geopandas/io/sql.py", line 161, in _read_postgis
>     chunksize=chunksize,
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/pandas/io/sql.py", line 501, in read_sql
>     chunksize=chunksize,
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/pandas/io/sql.py", line 1771, in read_query
>     cursor = self.execute(*args)
>   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/pandas/io/sql.py", line 1723, in execute
>     cur = self.con.cursor()
> AttributeError: 'function' object has no attribute 'cursor'
```
@KathiEsterl Can you help me?

**Comment by KathiEsterl** [comment]
> I tried to run your branch, but the task 'generate_pv_ground_mounted' fails with this error:
> 
> ```
> > 
> > [2021-06-24 13:57:25,697] {taskinstance.py:1150} ERROR - 'function' object has no attribute 'cursor'
> > Traceback (most recent call last):
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
> >     result = task_copy.execute(context=context)
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/airflow/operators/python_operator.py", line 113, in execute
> >     return_value = self.execute_callable()
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/airflow/operators/python_operator.py", line 118, in execute_callable
> >     return self.python_callable(*self.op_args, **self.op_kwargs)
> >   File "/home/clara/test-egon-data/git/eGon-data/src/egon/data/processing/pv_ground_mounted.py", line 849, in regio_of_pv_ground_mounted
> >     pv_rora, pv_agri, pv_per_distr, pv_rora_100RE, pv_agri_100RE, pv_per_distr_100RE = run_methodology(con = db.engine, path='', pow_per_area = 0.04, join_buffer = 10, max_dist_hv = 20000, show_map=True)
> >   File "/home/clara/test-egon-data/git/eGon-data/src/egon/data/processing/pv_ground_mounted.py", line 521, in run_methodology
> >     potentials_rora, potentials_agri = potential_areas(con, join_buffer)
> >   File "/home/clara/test-egon-data/git/eGon-data/src/egon/data/processing/pv_ground_mounted.py", line 118, in potential_areas
> >     potentials_rora = gpd.GeoDataFrame.from_postgis(sql, con)
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/geopandas/geodataframe.py", line 668, in from_postgis
> >     chunksize=chunksize,
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/geopandas/io/sql.py", line 161, in _read_postgis
> >     chunksize=chunksize,
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/pandas/io/sql.py", line 501, in read_sql
> >     chunksize=chunksize,
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/pandas/io/sql.py", line 1771, in read_query
> >     cursor = self.execute(*args)
> >   File "/home/clara/test-egon-data/venv/lib/python3.7/site-packages/pandas/io/sql.py", line 1723, in execute
> >     cur = self.con.cursor()
> > AttributeError: 'function' object has no attribute 'cursor'
> ```
> 
> @KathiEsterl Can you help me?

A bracket pair must have gone missing on the way - I am sorry for that! I corrected it, it should be working now.

**Comment by ClaraBuettner** [comment]
Thank you @KathiEsterl now the generation of both wind and pv farms works well.
@CarlosEpia and @KathiEsterl: I would like to ask you for some small adjustments before we merge this: 

- [ ] add a note to the [CHANGELOG](https://github.com/openego/eGon-data/blob/dev/CHANGELOG.rst)
- [ ] don't save intermediate results in files (I noticed that both of you created intermediate .csv, .geojson files, it's ok when you do this to check the results but I think it's not useful in the main branch. )
- [ ] you added some additional dependencies to the setup.py, all of them are already included in other packages, so they can be removed
- [ ]   'Codacy Static Code Analysis' shows some issues which can be solved automatically (e.g. with black which is also included in Spyder, please write if you need some help here)
- [ ] please add docstrings to your methods (at least for the central ones) which also describe the parameters (see e.g. here: https://github.com/openego/eGon-data/blob/9ffa8e3f4f3778fbe0e3d2586d9b82d286515677/src/egon/data/datasets/osm/__init__.py#L43-L52)

**Comment by KathiEsterl** [comment]
> Thank you @KathiEsterl now the generation of both wind and pv farms works well.
> @CarlosEpia and @KathiEsterl: I would like to ask you for some small adjustments before we merge this:
> 
>     * [x]  add a note to the [CHANGELOG](https://github.com/openego/eGon-data/blob/dev/CHANGELOG.rst)
> 
>     * [x]  don't save intermediate results in files (I noticed that both of you created intermediate .csv, .geojson files, it's ok when you do this to check the results but I think it's not useful in the main branch. )
> 
>     * [x]  you added some additional dependencies to the setup.py, all of them are already included in other packages, so they can be removed
> 
>     * [x]    'Codacy Static Code Analysis' shows some issues which can be solved automatically (e.g. with black which is also included in Spyder, please write if you need some help here)
> 
>     * [x]  please add docstrings to your methods (at least for the central ones) which also describe the parameters (see e.g. here: https://github.com/openego/eGon-data/blob/9ffa8e3f4f3778fbe0e3d2586d9b82d286515677/src/egon/data/datasets/osm/__init__.py#L43-L52
>       )

Thanks for your comments. All of the points above have been worked on and finished now.

**Review by ClaraBuettner — APPROVED** [review]
I tested your tasks (for Schleswig-Holstein and Germany) and everything runs successfully. As mentioned in #280 there is still the problem that the task import-hydro-biomass deletes all rows from the power plant table. I would like to merge this PR directly after #280.

File: src/egon/data/processing/wind_farms.py
Comment (nesnoj) [inline src/egon/data/processing/wind_farms.py:34]: @CarlosEpia @KathiEsterl @ClaraBuettner if you use GF 2,3 and 4, you still have some water bodies in the data, e.g. Flensburger Förde (cf. [VG250 docs](https://sg.geodatenzentrum.de/web_download/vg/vg250-ew_3112/vg250-ew_3112.pdf)).
As the RE potential data should not contain data in those areas, your results are most likely not affected but to be sure, use GF=4 to use land only..

**Pull request merged**
